spec/lib/whitehall_importer_spec.rb

```diff
@@ -1,30 +1,53 @@
 # frozen_string_literal: true
 
 RSpec.describe WhitehallImporter do
-  describe ".import" do
-    before { allow(WhitehallImporter::Import).to receive(:call) }
+  describe ".import_and_sync" do
+    before do
+      allow(ResyncService).to receive(:call)
+      allow(WhitehallImporter::ClearLinksetLinks).to receive(:call)
+    end
 
-    it "creates a WhitehallImport" do
-      expect { WhitehallImporter.import(build(:whitehall_export_document)) }
+    let(:whitehall_export_document) { build(:whitehall_export_document) }
+
+    it "creates and returns a WhitehallImport" do
+      whitehall_import = nil
+      expect { whitehall_import = WhitehallImporter.import_and_sync(whitehall_export_document) }
         .to change { WhitehallImport.count }
         .by(1)
+      expect(whitehall_import).to be_an_instance_of(WhitehallImport)
     end
 
-    it "imports a document" do
-      expect(WhitehallImporter::Import).to receive(:call)
-      WhitehallImporter.import(build(:whitehall_export_document))
+    it "stores the exported whitehall data" do
+      WhitehallImporter.import_and_sync(whitehall_export_document)
+      whitehall_import = WhitehallImport.find_by(whitehall_document_id: whitehall_export_document["id"])
+      expect(whitehall_import.payload).to eq(whitehall_export_document)
     end
 
-    it "stores the exported whitehall data" do
-      whitehall_export = build(:whitehall_export_document)
-      record = WhitehallImporter.import(whitehall_export)
-      expect(record.payload).to eq(whitehall_export)
+    it "doesn't sync if import fails" do
+      allow(WhitehallImporter::Import).to receive(:call)
+        .with(whitehall_export_document)
+        .and_raise(WhitehallImporter::AbortImportError, "Booo, import failed")
+
+      expect(ResyncService).not_to receive(:call)
+      expect(WhitehallImporter::ClearLinksetLinks).not_to receive(:call)
```

Comment:
You can probably simplify this to be that :sync doesn't get called.

Reply:
Good shout!